pgjdbc/src/main/java/org/postgresql/ssl/jdbc4/LibPQFactory.java

```diff
@@ -252,6 +258,31 @@ public boolean verify(String hostname, SSLSession session) {
     }
     // Extract the common name
     X509Certificate serverCert = peerCerts[0];
+
+    try {
+      // Check for Subject Alternative Names (see RFC 6125)
+      Collection subjectAltNames = serverCert.getSubjectAlternativeNames();
+
+      if (subjectAltNames != null) {
+
+        Iterator sanIt = subjectAltNames.iterator();
+
+        while (sanIt.hasNext()) {
+
+          List list = (List) sanIt.next();
+          String san = ((String) list.get(1));
```

Comment:
Extra braces, and `san`/`type` should probably be declared in a `type`, `san` order as per `getSubjectAlternativeNames` documentation